libs/src/services/solanaWeb3Manager/rewardsAttester.ts

```diff
@@ -610,6 +616,11 @@ export class RewardsAttester {
       )
 
       if (shouldReselect) {
+        // Add failing nodes to the blocklist, trimming out oldest nodes if necessary
+        this.discoveryNodeBlocklist = [
+          ...this.discoveryNodeBlocklist,
+          ...failingNodes
+        ].slice(-1 * MAX_DISCOVERY_NODE_BLOCKLIST_LEN)
```

Comment:
Does this ensure we move nodes out of the blocklist eventually?

Reply:
that's the idea - could have improved this with a time based pruning mechanism but I think this simple approach is sufficient for our needs